test/IntegrationTests/KafkaTests.cs

```diff
@@ -41,12 +44,32 @@ public void SubmitsTraces(string packageVersion)
         using var collector = new MockSpansCollector(Output);
         SetExporter(collector);
 
-        collector.Expect("OpenTelemetry.AutoInstrumentation.Kafka", span => span.Kind == Span.Types.SpanKind.Producer && ValidateProducerSpan(span, topicName, -1), "Produced without delivery handler.");
-        collector.Expect("OpenTelemetry.AutoInstrumentation.Kafka", span => span.Kind == Span.Types.SpanKind.Producer && ValidateProducerSpan(span, topicName, 0), "Produced with delivery handler.");
-        collector.Expect("OpenTelemetry.AutoInstrumentation.Kafka", span => span.Kind == Span.Types.SpanKind.Producer && ValidateProducerSpan(span, topicName, -1, true), "Produced a tombstone.");
+        // Failed produce attempts made before topic is created.
+        collector.Expect(KafkaInstrumentationScopeName, span => span.Kind == Span.Types.SpanKind.Producer && ValidateProduceExceptionSpan(span, topicName), "Failed Produce attempt with delivery handler set.");
+        collector.Expect(KafkaInstrumentationScopeName, span => span.Kind == Span.Types.SpanKind.Producer && ValidateProduceExceptionSpan(span, topicName), "Failed Produce attempt without delivery handler set.");
+        collector.Expect(KafkaInstrumentationScopeName, span => span.Kind == Span.Types.SpanKind.Producer && ValidateProduceExceptionSpan(span, topicName), "Failed ProduceAsync attempt.");
+
+        // For 1.4.0 null is returned when attempting to read from non-existent topic,
+        // and no exception is thrown,
+        // instrumentation creates no span in such case.
+        if (packageVersion == "2.3.0")
```

Comment:
packageVersion => 2.3.0?
With parsing by System.Version.

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3165/commits/9000fc80ff0761684ecf5df5b512aae386823374